diff --git a/paddle/fluid/inference/api/analysis_predictor.cc b/paddle/fluid/inference/api/analysis_predictor.cc
index 28f8233117..df8220b119 100644
--- a/paddle/fluid/inference/api/analysis_predictor.cc
+++ b/paddle/fluid/inference/api/analysis_predictor.cc
@@ -464,7 +464,9 @@ bool AnalysisPredictor::Init(
   paddle::platform::SetNumThreads(config_.cpu_math_library_num_threads());
 
   std::string model_path = config_.prog_file();
-  if (!model_path.empty()) {
+  if (config_.model_from_memory()) {
+    load_pir_model_ = config_.new_executor_enabled();
+  } else if (!model_path.empty()) {
     load_pir_model_ =
         model_path.substr(model_path.find_last_of(".") + 1) == "json";
   } else if (!config_.model_dir().empty()) {
@@ -1379,7 +1381,7 @@ bool AnalysisPredictor::SaveOrLoadPirParameters(bool for_save) {
       }
 
     } else {
-      if (std::filesystem::exists(config_.params_file())) {
+      if (config_.model_from_memory_ || std::filesystem::exists(config_.params_file())) {
         pir::LoadCombineFunction(config_.params_file(),
                                  filter_param_names,
                                  &tensor_out,
diff --git a/paddle/fluid/pir/serialize_deserialize/src/interface.cc b/paddle/fluid/pir/serialize_deserialize/src/interface.cc
index bc597db6d1..6c45afcd46 100644
--- a/paddle/fluid/pir/serialize_deserialize/src/interface.cc
+++ b/paddle/fluid/pir/serialize_deserialize/src/interface.cc
@@ -74,8 +74,14 @@ void WriteModule(const pir::Program& program,
 bool ReadModule(const std::string& file_path,
                 pir::Program* program,
                 int64_t pir_version) {
-  std::ifstream f(file_path);
-  Json data = Json::parse(f);
+  Json data;
+  if (file_path.size() < 1000) {
+    std::ifstream f(file_path);
+    data = Json::parse(f);
+  } else {
+    data = Json::parse(file_path);
+  }
+  
   if (pir_version < 0) {
     pir_version = DEVELOP_VERSION;
     VLOG(6) << "pir_version is null, get pir_version: " << pir_version;
diff --git a/paddle/fluid/pir/serialize_deserialize/src/save_load_parameters.cc b/paddle/fluid/pir/serialize_deserialize/src/save_load_parameters.cc
index 1d563c326a..0871d34170 100644
--- a/paddle/fluid/pir/serialize_deserialize/src/save_load_parameters.cc
+++ b/paddle/fluid/pir/serialize_deserialize/src/save_load_parameters.cc
@@ -12,6 +12,7 @@ limitations under the License. */
 #include <cstdint>
 #include <fstream>
 #include <numeric>
+#include <sstream>
 
 #include "glog/logging.h"
 #include "paddle/fluid/framework/lod_tensor.h"
@@ -188,13 +189,20 @@ void LoadCombineFunction(const std::string& file_path,
                          std::vector<phi::DenseTensor*>* out,
                          bool load_as_fp16,
                          phi::Place place) {
-  std::ifstream fin(file_path, std::ios::binary);
-  PADDLE_ENFORCE_EQ(static_cast<bool>(fin),
-                    true,
-                    common::errors::Unavailable(
-                        "Load operator fail to open file %s, please check "
-                        "whether the model file is complete or damaged.",
-                        file_path));
+  std::unique_ptr<std::istream> fin_ptr;
+  if (file_path.size() < 1000) {
+    fin_ptr = std::make_unique<std::ifstream>(file_path, std::ios::binary);
+    PADDLE_ENFORCE_EQ(static_cast<bool>(*fin_ptr),
+                      true,
+                      common::errors::Unavailable(
+                          "Load operator fail to open file %s, please check "
+                          "whether the model file is complete or damaged.",
+                          file_path));
+  } else {
+    fin_ptr = std::make_unique<std::istringstream>(
+        file_path, std::ios::in | std::ios::binary);
+  }
+  std::istream& fin = *fin_ptr;
 
   PADDLE_ENFORCE_GT(out->size(),
                     0UL,
